apps/src/lab2/views/components/Instructions/ResourcePanel/index.tsx

```diff
@@ -290,7 +290,19 @@ const ResourcePanel: React.FC<ResourcePanelProps> = ({
           headerClassName={headerClassName}
           rightHeaderContent={rightHeaderContent}
         >
-          {availableTabs[currentTab]}
+          <div className={styles.tabContentContainer}>
+            {getTypedKeys(availableTabs).map(tab => (
```

Comment:
I pulled your branch locally, and sometimes the chat history sometimes is not rendered in the AI tutor tab. 

This is on your branch locally:

<img width="1337" height="597" alt="render-all-tabs" src="https://github.com/user-attachments/assets/df8358a3-d009-4c4f-bfaa-61c201847a8a" />

This is on `staging` branch locally on the same level:

<img width="1340" height="563" alt="staging" src="https://github.com/user-attachments/assets/10b1007d-5899-45d9-947d-3d9364f5dbb3" />

Seems like there needs to be a wait?